src/librbd/cache/ImageCache.h

```diff
@@ -36,7 +36,7 @@ struct ImageCache {
   virtual void aio_discard(uint64_t offset, uint64_t length,
                            uint32_t discard_granularity_bytes,
                            Context *on_finish) = 0;
-  virtual void aio_flush(Context *on_finish) = 0;
+  virtual void aio_flush(io::FlushSource flush_source, Context *on_finish) = 0;
```

Comment:
To give more explanation about flush_source: This parameter is added to avoid IO hung issue when the cache is full and is flushing data on SSD to OSD, at the same time a write request from users is waiting because of no cache resource.    
When an internal aio_flush is called (eg, by ReplicatedWriteLog), it doesn't need to wait for all pending IOs are finished (eg, the IOs from users).